flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -221,22 +222,27 @@ public void shutDown() throws FlinkException {
 	/**
 	 * Creates and returns the task manager services.
 	 *
+	 * @param configuration Flink configuration.
 	 * @param taskManagerServicesConfiguration task manager configuration
 	 * @param taskManagerMetricGroup metric group of the task manager
 	 * @param resourceID resource ID of the task manager
 	 * @param taskIOExecutor executor for async IO operations
 	 * @param freeHeapMemoryWithDefrag an estimate of the size of the free heap memory
 	 * @param maxJvmHeapMemory the maximum JVM heap size
+	 * @param localCommunicationOnly True, to skip initializing the network stack.
+	 *                               Use only in cases where only one task manager runs.
 	 * @return task manager components
 	 * @throws Exception
 	 */
 	public static TaskManagerServices fromConfiguration(
+			Configuration configuration,
 			TaskManagerServicesConfiguration taskManagerServicesConfiguration,
 			MetricGroup taskManagerMetricGroup,
 			ResourceID resourceID,
 			Executor taskIOExecutor,
 			long freeHeapMemoryWithDefrag,
-			long maxJvmHeapMemory) throws Exception {
+			long maxJvmHeapMemory,
+			boolean localCommunicationOnly) throws Exception {
```

Comment:
I think whether it's local or not could be part of the `TaskManagerServicesConfiguration`.